Add optional config

ShMcK opened the pull request:
Resolves #146.

Allow the ability to overwrite axiosConfig for update & delete resource requests. See PR #146.

This allows the client to make requests at different endpoints.

In the example below, the endpoint for v2 is overwritten with v2.1.

const customConfig =  const config = {
  ...store.getState().api.endpoint.axiosConfig,
  baseURL: `https://host.com/v2.1/`, // rather than v2
};
dispatch(deleteResource(entity, customConfig));
Other axiosConfig properties could be overridden as well, such as host, timeout, etc.

egeriis commented:
@ShMcK Tests are failing and you've left a console.log statement in the code. Please revise :)

jonathonwalz commented:
Since this pull request is superseding #145, could we create the options object like below? Otherwise this will **not** provide the same functionality. I need a way to pass in data for the `meta` key.

```js
const options = {
  method: 'POST',
  data: JSON.stringify({
    data: resource
  }),
  ...axiosConfig,
};
```

ShMcK commented:
@egeriis, good idea. I've implemented a config override, rather than config replacement, in 6956c82. 

@jonathonwalz, I've also implemented the config override to fit your use case in 6956c82.

egeriis commented:
@jonathonwalz Good point! 

@ShMcK Good job, will take another look :)

jonathonwalz commented:
Since this hasn't been merged yet, I've run into a use-case where I need to modify the readResource request as well (modify the headers based on request). Would it make sense to include a similar change to allow modifying the axiosConfig for readResource too?

egeriis commented on src/jsonapi.js:
```diff
@@ -145,16 +148,17 @@ export const updateResource = (resource) => {
     dispatch(apiWillDelete(resource));
 
-    const { axiosConfig } = getState().api.endpoint;
+    const axiosConfig = getState().api.endpoint.axios;
```
There seem to be a typo here, `endpoint.axios` should be `endpoint.axiosConfig`. If this isn't caught be any tests, we should make that happen 😄

egeriis commented:
@ShMcK Sorry for letting this stall, I hope you don't mind revisiting this and revising this PR from my review ❤️

zdajani commented:
Just curious about what is happening with this PR I have a use case that really needs these changes and wondering if I can help move it along in any way.

kayodeadeniyi commented:
@ShMcK @egeriis I'd like to know if there is something I can do to help move this PR so that it can be merged
